cmd/recorder/upload_test.go

```diff
@@ -52,6 +52,22 @@ func TestUploadRecording(t *testing.T) {
 
 	rec.outPath = recFile.Name()
 
+	t.Run("invalid response", func(t *testing.T) {
+		middlewares = []middleware{
+			func(w http.ResponseWriter, r *http.Request) bool {
+				if r.URL.Path == "/plugins/com.mattermost.calls/bot/uploads" && r.Method == http.MethodPost {
+					w.WriteHeader(500)
+					fmt.Fprintln(w, `Internal Server Error`)
+					return true
+				}
+
+				return false
+			},
+		}
+		err := rec.uploadRecording()
+		require.EqualError(t, err, "failed to create upload: AppErrorFromJSON: model.utils.decode_json.app_error, body: Internal Server Error\n, invalid character 'I' looking for beginning of value")
```

Comment:
Just for my own clarity, I'm not sure how this error happened during the test? (the invalid character thing)

Reply:
@cpoile That's the internal `model.Client4` [logic](https://github.com/mattermost/mattermost/blob/master/server/public/model/utils.go#L340) trying to parse the `Internal Server Error` string as JSON.